### DISCUSSION TITLE
How to disable tons of warnings that I get with clang++ 18 on Debian 12 simdjson 3.10.1

### DISCUSSION THREAD

**Discussion by pmuvb** [simdjson/simdjson#2284] [General]
Hi,

I am getting tons of warnings coming from both header and source files. How do I disable them??

 /bitbucket/kskapp$ clang++ --version
Debian clang version 18.1.8 (++20240731024826+3b5b5c1ec4a3-1~exp1~20240731144843.145)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin

On Debian 12.

simdjson version 3.10.1 from August 26, 2024, it's in a "single" file. I just drop header and source files into my project.

I saw the     SIMDJSON_DISABLE_GCC_WARNING multiple statements in simdjson.h that 
start around line 460, but those look like apply to Visual Studio. Anything I add there just does not have 
any effect.

How do I disable all those warnings? I build my projects not to have any warnings.

Note, I don't want to disable warnings for my whole project but only those that are generated by simdjson.

Tx

**Comment by lemire** [comment]
Please do the following and report back on precisely what warning the compiler emits:

```
$ wget https://raw.githubusercontent.com/simdjson/simdjson/master/singleheader/simdjson.h https://raw.githubusercontent.com/simdjson/simdjson/master/singleheader/simdjson.cpp
$ clang++ -c simdjson.cpp -Wall -Wextra
```

If you are compiling with warning flags that are not included in `-Wall -Wextra`, then please note that we explicitly do not set it as a goal to avoid any kind of static analysis warning. These warnings are not enabled in these compilers for a reason: they are highly likely to provide false positives and create noise.

If, for some reason, you want to enable unusual warning flags in part of your software, but not others, you can do so. Please refer to the documentation of your compiler.

>  I don't want to disable warnings for my whole project but only those that are generated by simdjson.

Please refer to the documentation of your compiler.

**Reply by pmuvb** [reply]
My debug build: $ clang++ -std=c++23 -g3 -O0 -W'everything' -pedantic 

K, np, thanks

**Comment by lemire** [comment]
I am going to close this. It is always possible to find a static analysis tool that will produce warnings for almost any kind of code. It is not useful work to silence these warnings.

For example, with `-Weverything`, the following proper code generates a warning:

```C
int main(int argc, char **argv) {
    if (argc != 2) {
        return 0;
    }
    printf("argv[1] = %s\n", argv[1]);
}
```

(Code taken from StackOverflow.)

Indeed, the static analyser will consider that `argv[1]` is unsafe. Yet it is perfectly safe and standard code.

That's not reasonable at scale. It might be reasonable for one's own code, but it is not viable within simdjson to avoid pointer accesses.

We support compiling with `-Wall -Wextra` without warnings. If you need something like `-Wunsafe-buffer-usage` which generates false positives repeated, you should either accept the warnings in your logs, or silence selectively this warning. Please consult the documentation of your compiler.
